packages/components/select/src/useOption.ts

```diff
@@ -63,7 +72,7 @@ export function useOption(props, states) {
 
   const updateOption = (query: string) => {
     const regexp = new RegExp(escapeStringRegexp(query), 'i')
-    states.visible = regexp.test(currentLabel.value) || props.created
+    states.visible = regexp.test(currentLabel.value as string) || props.created
```

Comment:
`currentLabel: ComputedRef<string | number | boolean>`

How about using `String()` functions for conversion?